src/app/Plans/components/Wizard/SelectVMs.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+import { Pagination } from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  sortable,
+  compoundExpand,
+  classNames as classNamesTransform,
+  ICell,
+  IRow,
+  wrappable,
+} from '@patternfly/react-table';
+import tableStyles from '@patternfly/react-styles/css/components/Table/table';
+
+import { IVMs } from '../../types';
+import { useSortState, usePaginationState, useSelectionState } from '@app/common/hooks';
+import { StatusIcon, StatusType } from '@konveyor/lib-ui';
+
+interface ISelectVMsProps {
+  VMs: IVMs[];
+}
+
+const SelectVMs: React.FunctionComponent<ISelectVMsProps> = ({ VMs }: ISelectVMsProps) => {
+  const getSortValues = (vm: IVMs) => {
+    return ['', vm.Name, vm.Datacenter, vm.Cluster, vm.Host, vm.FolderPath, ''];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(VMs, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  React.useEffect(() => setPageNumber(1), [sortBy, setPageNumber]);
+
+  const { selectedItems, toggleItemSelected, areAllSelected, selectAll } = useSelectionState<IVMs>(
+    sortedItems
+  );
+
+  const { selectedItems: expandedVMs, toggleItemSelected: toggleVMExpanded } = useSelectionState<
+    IVMs
+  >(sortedItems);
+
+  const columns: ICell[] = [
+    {
+      // Using a custom column instead of Table's onSelect prop due to issues
```

Comment:
What issues came up here?

Reply:
That's duplicated from my comment in the providers table. Basically, there is a bug in Table where if you use `onSelect` together with the expandable row, it adds an empty column to the left of your expandable row because of its code for adding checkbox columns. This indents the expanded content by a bunch. I plan to open that as an issue in PF and we can eventually switch back to onSelect. We should do that because I'm pretty sure doing this breaks mobile friendliness.

Reply:
Got it, thanks!